cmd/generate/flag.go

```diff
@@ -75,6 +81,9 @@ func (f *flag) Validate() error {
 	if (f.VaultSecretName == "" || f.VaultSecretNamespace == "") && f.VaultSecretName != f.VaultSecretNamespace {
 		return microerror.Maskf(invalidFlagError, "you have to specify both or neither %q and %q", flagVaultSecretName, flagVaultSecretNamespace)
 	}
+	if f.SOPSKeysSource != "local" && f.SOPSKeysSource != "kubernetes" {
```

Comment:
Is `f.flagSOPSKeysDir` required if `f.SOPSKeysSource` == "local"? Or does `""` mean working directory.

Reply:
It isn't, it means then user runs SOPS against his default keychains (or rather SOPS-default keychains). As a matter of fact, dir is not required in any scenario.

`f.flagSOPSKeysDir` = "" and `f.flagSOPSKeysSource` is:
* `kubernetes` -> keys are downloaded to a tmp location which is then deleted
* `local` -> user runs against his local key chain

Reply:
Makes sense now, thanks.